src/main/java/com/contentful/java/cma/ModuleEnvironments.java

```diff
@@ -123,16 +123,36 @@ public CMAEnvironment branch(CMAEnvironment sourceEnvironment, CMAEnvironment ne
 
     try {
       if (environmentId == null) {
-        return service.branch(spaceId, sourceEnvironment.getId(), newEnvironment).blockingFirst();
+        return service.clone(spaceId, sourceEnvironment.getId(), newEnvironment).blockingFirst();
       } else {
-        return service.branch(spaceId, sourceEnvironment.getId(), environmentId, newEnvironment)
+        return service.clone(spaceId, sourceEnvironment.getId(), environmentId, newEnvironment)
             .blockingFirst();
       }
     } finally {
       newEnvironment.setSystem(system);
     }
   }
 
+  /**
+   * Create an environment based on the content of another environment.
+   * <p>
+   * This method will override the configuration specified through
+   * {@link CMAClient.Builder#setSpaceId(String)}.
+   *
+   * @param sourceEnvironment the environment to be taken as a source of branching
+   * @param newEnvironment    the environment to be created, based on the source.
+   * @return {@link CMAEnvironment} result instance
+   * @throws IllegalArgumentException if sourceEnvironment is null.
+   * @throws IllegalArgumentException if newEnvironment is null.
+   * @throws IllegalArgumentException if the space of the source environment is not set.
+   * @deprecated This method has been renamed to {@link #clone(CMAEnvironment, CMAEnvironment)}.
+   * It will be removed in the next major release.
+   */
+  @Deprecated
+  public CMAEnvironment branch(CMAEnvironment sourceEnvironment, CMAEnvironment newEnvironment) {
```

Comment:
This method does not seem to be tested. Can you add a test to see if the redirection does what it should. (I would take a simple environment clone test and call `.branch()` on it)